### ISSUE TITLE
Function descriptions - lg_dataset.py

### ISSUE THREAD

**Issue by sautee** [KeiLongW/battery-state-estimation#8]
First, thank you for all your work in putting out this excellent repository! This has been quite insightful.

I'm reviewing the lg_dataset.py data processing module. I understand some of the functions, but I'm not clear on the others. 

1. _get_data - computes SOC, assigns Voltage, Current, Temperature to 'x', SOC labels to 'y'
2. _time_string_to_seconds - converts the 'Program Time' column to seconds
3. get_discharge_whole_cycle - creates the test and train datasets using the '_get_data' function
4. get_stateful_cycle - not sure what this does
5. _to_padded_cycle - not sure what this does
6. _split_cycle - not sure what this does
7. _split_to_multiple_step - not sure what this does
8. keep_only_y_end - not sure what this does

Is my understanding of these functions correct and could you help describe the ones I'm unsure about?

Overall, it looks like you can process the TRAIN and TEST datasets in two ways - as (1) 'get discharge multiple step' or as (2) 'stateful cycle'. What does stateful cycle mean, and are there scenarios where you use one over the other?

Thanks.

**Comment by KeiLongW** [comment]
`_split_to_multiple_step`: split cycles to multiple sub-cycles with `steps` length
> e.g. `steps=5` [1,2,3,4,5,6,7,8,9,10] => [[1,2,3,4,5],[6,7,8,9,10]]

`keep_only_y_end`: in our experiments of LG data, we use $n$ inputs to estimate $1$ output (refer to paper section 3.3). So this function simply remove all steps in $y$ and keep the last step.

`_to_padded_cycle`: pad the cycles with $0$ so that all cycles have the same length
> e.g. [1,2] => [1,2,0], [1,2,3]=>[1,2,3], [1] => [1,0,0]

`get_stateful_cycle`, `_split_cycle`: used to prepare data for [stateful LSTM](https://keras.io/api/layers/recurrent_layers/lstm/) test. The paper doesn't include it, but you can refer to `lstm_soc_percentage_lg_stateful.ipynb` for details.

**Comment by sautee** [comment]
thanks @KeiLongW, much appreciated!

**Comment by sautee** [comment]
@KeiLongW : another quick question. I notice you include only the discharge cycles as part of model training. Any reason you don't include the corresponding charge cycles?

**Comment by KeiLongW** [comment]
@sautee  To the best of my understanding, SOC estimation in the discharge cycle is more challenging and essential because:
1) the battery discharge pattern is variable and dynamic. Unlike the charge cycle that is usually charged with a constant pattern such as CCCV.
2) in the scenario of EV, the drivers evaluate their route plan based on the estimated SOC during operation (discharge)

**Comment by sautee** [comment]
Understood, and makes sense. Thanks @KeiLongW !
